### PR TITLE
[Package Signing] Fix idempotency

### PR DESCRIPTION
**Pull request by loic-sharma**
Previously, the validate package signature job was designed such that it would fail validation requests for  packages that had already been validated unless the validation request was from a revalidation gesture. The idea was that this would make the job "resilient" to unintended state changes caused by message duplication. Essentially, this could happen:

1. Orchestrator sends a `Validate package X's signature` message. Validation passes.
2. Orchestrator resends the `Validate package X's signature` message. Validation fails (even though the signature is still valid).
3. Orchestrator sends a `Revalidate package X's signature` message. Validation passes.

This turned out to be a bad idea as the orchestrator may validate the same package multiple times if it was uploaded multiple times concurrently. With this PR, the new results would be:

1. Orchestrator sends a `Validate package X's signature` message. Validation passes.
2. Orchestrator resends the `Validate package X's signature` message. Validation passes.
3. Orchestrator sends a `Revalidate package X's signature` message. Validation passes.

### CODE REVIEW

File: src/Validation.PackageSigning.ExtractAndValidateSignature/Job.cs [hunk 1]
```diff
@@ -166,7 +169,7 @@ private void ConfigureJobServices(IServiceCollection services, IConfigurationRoo
                 });
 
                 client.Timeout = configurationRoot.GetValue<TimeSpan>(PackageDownloadTimeoutName);
-                client.DefaultRequestHeaders.Add("user-agent", $"{assemblyName.Name}/{assemblyName.Version}");
+                client.DefaultRequestHeaders.Add("user-agent", $"{assemblyName}/{assemblyVersion}");
```
Comment (skofman1) [inline src/Validation.PackageSigning.ExtractAndValidateSignature/Job.cs]: why do we need a specific user agent for the job?

Comment (joelverhagen) [reply src/Validation.PackageSigning.ExtractAndValidateSignature/Job.cs]: This was feedback I gave. This allows us a) debugging from the storage logs side and b) allows us to filter stats if these requests some day go through CDN (which they don't today, package URLs orchestrator generates are directly to blob storage).

File: tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests.csproj [hunk 2]
```diff
@@ -61,5 +61,8 @@
       <Name>Validation.PackageSigning.ExtractAndValidateSignature</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (xavierdecoster) [inline tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests.csproj]: What's this?

Comment (loic-sharma) [reply tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests.csproj]: AFAIK, this is a no-op that Visual Studio uses to mark a project as a test project.

**Pull request merged**